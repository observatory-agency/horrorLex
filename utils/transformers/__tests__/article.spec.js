const articleTransformer = require('../article');

const mockArticle = {
  Title: 'Shadowplays: Tod Browning\'s "Dracula" and Karl Freund\'s "The Mummy"',
  Author: 'Le Cain, Maximilian',
  Publication: '16:9 filmtidsskrift',
  Year: '2005',
  'DOI/Stable URL': 'N/A',
  'Free Online Copy': 'http://www.16-9.dk/2005-04/side11_inenglish.htm',
  'MLA 8 Citation': 'Le Cain, Maximilian. "Shadowplays: Tod Browning\'s \'Dracula\' and Karl Freund\'s \'The Mummy\'." 16:9 filmtidsskrift, vol. 3, no. 11, Apr. 2005, www.16-9.dk/2005-04/side11_inenglish.htm.',
  'Reprinted In': 'N/A',
  MUSE: 'No',
  JSTOR: 'No',
  'Horror Lex Summary': 'Coming Soon',
  Films: 'Dracula (1931)\nThe Mummy (1932)',
  Tags: 'vampires, Dracula, mummy, classics',
  Rec: '',
  Photo: '169F0011',
};

describe('articleTransformer', () => {
  it('should match the snapshot', () => {
    expect(articleTransformer(mockArticle)).toMatchSnapshot();
  });
});
