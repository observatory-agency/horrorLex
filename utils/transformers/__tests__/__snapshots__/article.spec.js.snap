// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleTransformer should match the snapshot 1`] = `
Object {
  "author": "Le Cain, Maximilian",
  "doiStableUrl": "N/A",
  "filmsDiscussed": Array [
    "Dracula (1931)",
    "The Mummy (1932)",
  ],
  "freeOnlineCopy": "http://www.16-9.dk/2005-04/side11_inenglish.htm",
  "horrorLexSummary": "Coming Soon",
  "jstor": "No",
  "mla8Citation": "Le Cain, Maximilian. \\"Shadowplays: Tod Browning's 'Dracula' and Karl Freund's 'The Mummy'.\\" 16:9 filmtidsskrift, vol. 3, no. 11, Apr. 2005, www.16-9.dk/2005-04/side11_inenglish.htm.",
  "muse": "No",
  "photo": "169F0011",
  "publication": "16:9 filmtidsskrift",
  "recommended": "",
  "reprintedIn": "N/A",
  "slug": "shadowplays-tod-brownings-dracula-and-karl-freunds-the-mummy",
  "tags": Array [
    "vampires",
    "Dracula",
    "mummy",
    "classics",
  ],
  "title": "Shadowplays: Tod Browning's \\"Dracula\\" and Karl Freund's \\"The Mummy\\"",
  "type": "article",
  "year": "2005",
}
`;
